api/v1alpha1/live_types_test.go

```diff
@@ -0,0 +1,38 @@
+package v1alpha1
+
+import (
+	"testing"
+	"time"
+
+	"gotest.tools/v3/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestLiveBackoffRemaining(t *testing.T) {
+	live := Live{
+		ObjectMeta: metav1.ObjectMeta{
+			Generation: 1,
+		},
+	}
+
+	live.SetPhase(LivePhase{Name: LivePhaseSucceeded})
+	currentTime := time.Now()
+	live.GetReadyCondition().LastTransitionTime = metav1.NewTime(currentTime)
+	assert.Equal(t, live.backoffRemainingAt(currentTime), time.Second*0)
+	live.Generation += 1
+
+	live.SetPhase(LivePhase{Name: LivePhaseApplying})
+	live.SetPhase(LivePhase{Name: LivePhaseFailed})
+	currentTime = currentTime.Add(time.Millisecond * 350)
+	live.GetReadyCondition().LastTransitionTime = metav1.NewTime(currentTime)
+	assert.Equal(t, live.backoffRemainingAt(currentTime), time.Millisecond*2000)
+
+	live.SetPhase(LivePhase{Name: LivePhaseApplying})
+	live.SetPhase(LivePhase{Name: LivePhaseFailed})
+	currentTime = currentTime.Add(time.Millisecond * 350)
+	live.GetReadyCondition().LastTransitionTime = metav1.NewTime(currentTime)
+	assert.Equal(t, live.backoffRemainingAt(currentTime), time.Millisecond*4000)
+
+	currentTime = currentTime.Add(time.Millisecond * 500)
+	assert.Equal(t, live.backoffRemainingAt(currentTime), time.Millisecond*3500)
+}
```

Comment:
Also `currentTime.Add(time.Second * 10)` and check that backoffRemainingAt is 0 to check when after expected time, not only exactly at it as the first case checks..